src/cpp/rediscluster.cpp

```diff
@@ -126,6 +126,43 @@ CommandReply RedisCluster::run(AddressAnyCommand &cmd)
     return _run(cmd, _last_prefix);
 }
 
+// Run a non-keyed Command that addresses any db node on the server
+CommandReply RedisCluster::run(AddressAllCommand &cmd)
+{
+    // Bounds check the key_index
+    if (cmd.key_index != -1 && cmd.get_field_count() < cmd.key_index) {
+        throw SRInternalException("Invalid key_index executing command!");
+    }
+
+    // Find the segment to be tweaked for each node
+    std::string field;
+    if (cmd.key_index != -1) {
+        Command::const_iterator it = cmd.cbegin();
+        int i = 0;
+        for ( ; it != cmd.cend(); it++) {
+            if (i == cmd.key_index)
+                field = std::string(it->data(), it->size());
+            i++;
+        }
+    }
+
+    // Loop through all nodes to execute the command at each
+    std::vector<DBNode>::const_iterator node = _db_nodes.cbegin();
+    CommandReply reply;
+    for ( ; node != _db_nodes.cend(); node++) {
+        // swap in a replacement segment for each one
+        std::string new_field = "{" + node->prefix + "}." + field;
+        cmd.set_field_at(new_field, cmd.key_index, true);
+
+        // Execute the updated command
+        cmd.set_exec_address_port(node->ip, node->port);
+        reply = _run(cmd, node->prefix);
+        if (reply.has_error() > 0)
+            break; // Short-circuit failure on error
```

Comment:
Why break and not just throw the error here?

Reply:
Largely stylistic. I wanted to keep as much extraneous stuff out of the loop as possible to make it easy to read what normal (successful execution) path is doing.  With just a single return statement, yes, I could have done that, but if we ever go back and add more stuff in the case of a failure, it will be more complex.